packages/cli/src/commands/push.ts

```diff
@@ -141,23 +144,37 @@ async function collectFilesToUpload(entrypoint: string) {
 
   if (config.configFile) {
     files.push(getFileEntry(config.configFile));
+    if (fs.existsSync('package.json')) { files.push(getFileEntry('package.json')); }
+    if (fs.existsSync(IGNORE_FILE)) { files.push(getFileEntry(IGNORE_FILE)); }
     if (config.referenceDocs.htmlTemplate) {
       const dir = getFolder(config.referenceDocs.htmlTemplate);
       const fileList = getFilesList(dir, []);
       files.push(...fileList.map(f => getFileEntry(f)));
     }
     if (config.rawConfig && config.rawConfig.lint && config.rawConfig.lint.plugins) {
+      let pluginFiles = new Set();
       for (const plugin of config.rawConfig.lint.plugins) {
         if (typeof plugin !== 'string') continue;
-        files.push(getFileEntry(plugin));
+        const fileList = getFilesList(getFolder(plugin), []);
+        fileList.map(f => pluginFiles.add(f));
       }
+      files
+        .push(...(filterFilesByExtensions(Array.from(pluginFiles) as string[]))
+        .map(f => getFileEntry(f)));
```

Comment:
Fix formatting here:

```suggestion
      files.push(
          ...filterFilesByExtensions(Array.from(pluginFiles)).map(f => getFileEntry(f))
        );
```